File: Cpp/Problem_1347.cpp
#include <iostream>
#include <string>
#include <vector>

using namespace std;

// Problem - 1347: https://leetcode.com/problems/minimum-number-of-steps-to-make-two-strings-anagram/
// C++ Solution!
class Solution {
public:
    int minSteps(string s, string t) {
        vector<int> charCount(26, 0);

        for (int i = 0; i < s.size(); i++) {
            charCount[s[i] - 'a']++;
            charCount[t[i] - 'a']--;
        }

        int res = 0;
        for (int i = 0; i < 26; i++) {
            res += charCount[i] > 0 ? charCount[i] : 0;
        }

        return res;
    }
};

int main() {
    Solution sol;

    cout << sol.minSteps("bab", "aba") << endl;
    cout << sol.minSteps("leetcode", "practice") << endl;
    cout << sol.minSteps("anagram", "mangaar") << endl;

    return 0;
}

File: Python/Problem_1347.py
# Problem - 1347: https://leetcode.com/problems/minimum-number-of-steps-to-make-two-strings-anagram/
# Python Solution!
class Solution:
    def minSteps_regular(self, s: str, t: str) -> int:
        charCount = [0] * 26
        for i in range(len(s)):
            charCount[ord(s[i]) - ord('a')] += 1
            charCount[ord(t[i]) - ord('a')] -= 1
        res = 0
        for i in range(26):
            res += charCount[i] if charCount[i] > 0 else 0
        return res

    def minSteps(self, s: str, t: str) -> int:
        res = 0
        for char in set(s):
            diff = s.count(char) - t.count(char)
            res += diff if diff > 0 else 0
        return res


if __name__=='__main__':
    sol = Solution()

    print(sol.minSteps("bab", "aba"))
    print(sol.minSteps("leetcode", "practice"))
    print(sol.minSteps("anagram", "mangaar"))